specs/configuration_files_specification.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Configuration Files Specification
+layout: posts
+version: 1
+---
+
+# Configuration Files Specification
+
+## Introduction
+
+Various specifications attempt to define configuration files and file formats. This
+specification establishes where these files should be looked for, in which
+order, and how precedence, masking, extensions and overrides work.
+
+The purpose of the rules defined here is to allow OS vendors to implement the
+hermetic-usr pattern, where all vendor files are shipped in the vendor tree itself
+(`/usr/`), including configuration files with system defaults, while allowing local
+or vendor overrides without modifying the original files, for easier management. This is
+especially beneficial for image-based deployments, where the vendor tree is read-only.
+
+These rules are derived from existing real-world usage from the [systemd
+project](https://github.com/systemd/systemd) and the [libeconf
+project](https://github.com/openSUSE/libeconf).
+
+This specification is agnostic toward the actual format and content of the configuration
+files, the precise path used under each top-level hierarchy, and also toward the
+filenames and extensions, with the exception of the `.d/` notation for drop-ins
+directories. It is strongly encouraged to enforce a specific suffix for the configuration
+files, in order to disambiguate (e.g.: with backup files), but the choice of which suffix
+to use is left to each implementation.
+
+## Storage Directories and Overrides
+In order to allow shipping system defaults owned by the OS vendor, while at the same
+time letting local users or admins override those defaults, `/usr/` and `/etc/` are both
+supported for storage of configuration files, with the latter having higher priority.
+Programs must work correctly if no configuration files are found in `/etc/`.
+Optionally, `/run/` is also supported for ephemeral overrides.
```

Comment:
Another thing that came up last year at the summit was, that systemd also supports `/usr/local/lib` for units with precedence in between `/usr/lib/` and `/run/`. It was decided to leave this out because some people did `/usr/lib/etc` (or was it `/usr/share/etc`?). Also mentioned, while not strictly config, at least systemd generators have search paths where `/run/` comes before `/etc/` and the runtime config they generate can have higher precedence than `/etc/` and lower precedece than `/usr/lib/` and systemctl drops transient runtime overrides into `/run/systemd/system.control` that have higher priority than the config in `/etc/systemd/system`.

These are all special cases of one specific project, but other people might have similar needs, so should we maybe give some guidance here, so that if people (probably wrongly) think they have the need to wedge something in between `/usr/lib/` and `/etc/` or (maybe more realistically) need runtime config that trumps config in `/etc/` they arrive at the some solutions as others?

Reply:
That sounds way too specific to me, the purpose of this is to tell people "use libeconf if you don't have anything, and if you really really have to do your own thing, follow these basic rules so that we can get a stateless system"